### PR TITLE
api: add new progression for waiting on user cleanup

### PR DESCRIPTION
**Pull request by raghavendra-talur**
Signed-off-by: Raghavendra Talur <raghavendra.talur@gmail.com>

### CODE REVIEW

**Comment by raghavendra-talur** [comment]
Sample output
```
$ k get drpc -o wide -w
NAME          AGE   PREFERREDCLUSTER   FAILOVERCLUSTER   DESIREDSTATE   CURRENTSTATE   PROGRESSION   START TIME             DURATION   PEER READY
test-1-drpc   18h   rdr-dr1            rdr-dr2           Failover       FailedOver     Cleaning Up   2024-05-12T14:22:12Z              False
test-1-drpc   18h   rdr-dr1            rdr-dr2           Failover       FailedOver     Cleaning Up   2024-05-12T14:22:12Z              False
test-1-drpc   18h   rdr-dr1            rdr-dr2           Failover       FailedOver     Cleaning Up   2024-05-12T14:22:12Z              False
test-1-drpc   18h   rdr-dr1            rdr-dr2           Failover       FailedOver     Cleaning Up   2024-05-12T14:22:12Z              False
test-1-drpc   18h   rdr-dr1            rdr-dr2           Failover       FailedOver     Cleaning Up   2024-05-12T14:22:12Z              False
test-1-drpc   19h   rdr-dr1            rdr-dr2           Failover       FailedOver     Cleaning Up   2024-05-12T14:22:12Z              False
test-1-drpc   19h   rdr-dr1            rdr-dr2           Failover       FailedOver     Cleaning Up   2024-05-12T14:22:12Z              False
test-1-drpc   19h   rdr-dr1            rdr-dr2           Failover       FailedOver     WaitOnUserToCleanUp   2024-05-12T14:22:12Z              False
test-1-drpc   19h   rdr-dr1            rdr-dr2           Failover       FailedOver     WaitOnUserToCleanUp   2024-05-12T14:22:12Z              False
test-1-drpc   19h   rdr-dr1            rdr-dr2           Failover       FailedOver     WaitOnUserToCleanUp   2024-05-12T14:22:12Z              False
test-1-drpc   19h   rdr-dr1            rdr-dr2           Failover       FailedOver     WaitOnUserToCleanUp   2024-05-12T14:22:12Z              False
test-1-drpc   19h   rdr-dr1            rdr-dr2           Failover       FailedOver     WaitOnUserToCleanUp   2024-05-12T14:22:12Z              False
test-1-drpc   19h   rdr-dr1            rdr-dr2           Failover       FailedOver     WaitOnUserToCleanUp   2024-05-12T14:22:12Z              False
test-1-drpc   19h   rdr-dr1            rdr-dr2           Failover       FailedOver     WaitOnUserToCleanUp   2024-05-12T14:22:12Z              False
test-1-drpc   19h   rdr-dr1            rdr-dr2           Failover       FailedOver     WaitOnUserToCleanUp   2024-05-12T14:22:12Z              False
test-1-drpc   19h   rdr-dr1            rdr-dr2           Failover       FailedOver     WaitOnUserToCleanUp   2024-05-12T14:22:12Z              False
test-1-drpc   19h   rdr-dr1            rdr-dr2           Failover       FailedOver     WaitOnUserToCleanUp   2024-05-12T14:22:12Z              False
test-1-drpc   19h   rdr-dr1            rdr-dr2           Failover       FailedOver     WaitOnUserToCleanUp   2024-05-12T14:22:12Z              False
test-1-drpc   19h   rdr-dr1            rdr-dr2           Failover       FailedOver     WaitOnUserToCleanUp   2024-05-12T14:22:12Z              False
test-1-drpc   19h   rdr-dr1            rdr-dr2           Failover       FailedOver     WaitOnUserToCleanUp   2024-05-12T14:22:12Z              False
test-1-drpc   19h   rdr-dr1            rdr-dr2           Failover       FailedOver     WaitOnUserToCleanUp   2024-05-12T14:22:12Z              False
test-1-drpc   19h   rdr-dr1            rdr-dr2           Failover       FailedOver     WaitOnUserToCleanUp   2024-05-12T14:22:12Z              False
test-1-drpc   19h   rdr-dr1            rdr-dr2           Failover       FailedOver     WaitOnUserToCleanUp   2024-05-12T14:22:12Z              False
test-1-drpc   19h   rdr-dr1            rdr-dr2           Failover       FailedOver     WaitOnUserToCleanUp   2024-05-12T14:22:12Z              False
test-1-drpc   19h   rdr-dr1            rdr-dr2           Failover       FailedOver     WaitOnUserToCleanUp   2024-05-12T14:22:12Z              False
test-1-drpc   19h   rdr-dr1            rdr-dr2           Failover       FailedOver     WaitOnUserToCleanUp   2024-05-12T14:22:12Z              False
test-1-drpc   19h   rdr-dr1            rdr-dr2           Failover       FailedOver     Cleaning Up           2024-05-12T14:22:12Z              False
test-1-drpc   19h   rdr-dr1            rdr-dr2           Failover       FailedOver     Completed             2024-05-12T14:22:12Z   15h18m23.34414583s   True
test-1-drpc   19h   rdr-dr1            rdr-dr2           Failover       FailedOver     Completed             2024-05-12T14:22:12Z   15h18m23.34414583s   True
test-1-drpc   19h   rdr-dr1            rdr-dr2           Failover       FailedOver     Completed             2024-05-12T14:22:12Z   15h18m23.34414583s   True
test-1-drpc   19h   rdr-dr1            rdr-dr2           Relocate       FailedOver     Completed             2024-05-12T14:22:12Z   15h18m23.34414583s   True
test-1-drpc   19h   rdr-dr1            rdr-dr2           Relocate       Initiating     PreparingFinalSync    2024-05-13T05:42:00Z                        True
test-1-drpc   19h   rdr-dr1            rdr-dr2           Relocate       Relocating     RunningFinalSync      2024-05-13T05:42:00Z                        True
test-1-drpc   19h   rdr-dr1            rdr-dr2           Relocate       Relocating     WaitOnUserToCleanUp   2024-05-13T05:42:00Z                        False
```

**Review by ShyamsundarR — COMMENTED** [review]
Quick comment, update comment above `setProgression` to include this progression under the appropriate categories.

**Comment by raghavendra-talur** [comment]
> Quick comment, update comment above `setProgression` to include this progression under the appropriate categories.

Done

File: controllers/drplacementcontrol.go [hunk 1]
```diff
@@ -1865,6 +1865,11 @@ func (d *DRPCInstance) ensureVRGManifestWorkOnClusterDeleted(clusterName string)
 	}
 
 	d.log.Info("Request not complete yet", "cluster", clusterName)
+
+	if d.instance.Spec.ProtectedNamespaces != nil && len(*d.instance.Spec.ProtectedNamespaces) > 0 {
```
Comment (ShyamsundarR) [inline controllers/drplacementcontrol.go:1869]: We need to handle something similar in `cleanupForVolSync` which is Volsync specifc?

Comment (raghavendra-talur) [reply controllers/drplacementcontrol.go:1869]: I would like to handle that in #1391 and merge this for now as this fixes RBD.

**Review by ShyamsundarR — APPROVED** [review]

**Pull request merged**